hell/adventcalendar/2025/13/index.md

```diff
@@ -1,35 +1,440 @@
+  <msup>
+    <mi>a</mi>
+    <mn>2</mn>
+  </msup>
+  <mo>+</mo>
+  <msup>
+    <mi>b</mi>
+    <mn>2</mn>
+  </msup>
+  <mo>=</mo>
+  <msup>
+    <mi>c</mi>
+    <mn>2</mn>
+  </msup>
+</math>
+```
+
+Unlike plain HTML with <code>sup</code> or <code>span</code>, which only describe presentation, MathML defines each role explicitly:
+
+- <code>math</code> represents the entire mathematical expression.
+- <code>msup</code> defines a superscript relationship (a base and an exponent).
+- <code>mi</code> is a mathematical identifier, typically a variable such as <strong>a</strong>, <strong>b</strong>, or <strong>c</strong>.
+- <code>mn</code> is a mathematical number, like <strong>2</strong>.
+- <code>mo</code> is a mathematical operator, such as <strong>+</strong> or <strong>=</strong>.
+
+Other alternatives, like the ones mentioned above, may offer similar capabilities, but they typically rely on an assistive or hidden MathML layer. In practice, MathML remains the only web-standard markup that expresses mathematical roles natively in the DOM.
+
+With this approach, the accessibility tree shows good semantics and VoiceOver knows well what to do.
+
+<img alt="Accessibility tree view of a MathML formula showing nested semantic elements. The tree includes nodes such as MathMLMath, MathMLSup, MathMLIdentifier, MathMLNumber, and MathMLOperator, representing the structure of the equation a² + b² = c²." src="./mathml-a11y-tree.png" 
+/>
+
+However, and this is something we will see through the whole article, screen reader support for the <code>math</code> tag is differently depending on the assistive technology. VoiceOver seems to be doing a pretty good job, [JAWS also makes it easy for both speech and braille](https://www.freedomscientific.com/training/teachers/accessing-math-content-with-jaws-and-fusion/), and [NVDA needs an add-on to make it work called MathCat](https://github.com/nvaccess/nvda/issues/17667) because if not, the <code>math</code> tag will be ignored. A major pull request ([#18323](https://github.com/nvaccess/nvda/pull/18323)) was merged on 17 November 2025 which integrates MathCAT into NVDA core, meaning users won’t have to find/install a separate add-on to handle math.
+
+<section aria-labelledby="example-formula">
+  <h3 id="example-formula">How screen readers interpret the formula</h3>
+  <details>
+    <summary>NVDA + Firefox (Windows with MathCAT add-on)</summary>
+    region eigh squared plus b squared is equal to c squared space
+  </details>
+  <details>
+    <summary>VoiceOver + Safari (Mac)</summary>
+    a squared + b squared = c squared, with 5 items, maths
+  </details>
+  <details style="margin-bottom: 2rem;">
+    <summary>VoiceOver + Safari (iOS)</summary>
+    a squared plus b squared equals c squared, Math
+  </details>
+</section>
+
+Let's look at a more complicated case. Instead of just the way to display the formula, let's see how to actually prove it and how that will be announced by screen readers.
```

Comment:
```suggestion
Let's look at a more complicated case. Instead of just displaying the formula, let's see how to actually prove it and how screen readers will announce it.
```